src/python/WMCore/WMSpec/Steps/Executors/LogArchive.py

```diff
@@ -127,7 +143,25 @@ def execute(self, emulator=None, **overrides):
                     'GUID': None
                     }
 
+        eosFileInfo = {'LFN': self.getEOSLogLFN(),
+                       'PFN': tarBallLocation,
+                       'PNN': None,
+                       'GUID': None
+                       }
         signal.signal(signal.SIGALRM, alarmHandler)
+        signal.alarm(waitTime)
+        try:
+            eosmanager(eosFileInfo)
+            eosServerPrefix = eosStageOutParams['lfn-prefix'].replace("root://eoscms.cern.ch//eos/cms",
```

Comment:
Isn't `eosFileInfo` structure updated in place? I think `eosFileInfo['PFN']` will contain the final file location. We could then pass it to the `setLogURL` and let that method replace the prefix.

Reply:
It is not in place replace. It returns new string.  The reason I put the code here is that I don't wan t to assume the prefix in general although it won't change. I think we have to think about more where is the best place, but since it is the closest place actual location is set, I just put it here.